spec-dtslint/operators/combineLatest-spec.ts

```diff
@@ -0,0 +1,110 @@
+      const a = of(1, 2, 3);
+      const b = ['a', 'b', 'c'];
+      const c = ['d', 'e', 'f'];
+      const d = ['g', 'h', 'i'];
+      const e = ['j', 'k', 'l'];
+      const res = a.pipe(combineLatest(b, c, d, e)); // $ExpectType Observable<[number, string, string, string, string]>
+    });
+
+    it('should infer correctly with 5 params', () => {
+      const a = of(1, 2, 3);
+      const b = ['a', 'b', 'c'];
+      const c = ['d', 'e', 'f'];
+      const d = ['g', 'h', 'i'];
+      const e = ['j', 'k', 'l'];
+      const f = ['m', 'n', 'o'];
+      const res = a.pipe(combineLatest(b, c, d, e, f)); // $ExpectType Observable<[number, string, string, string, string, string]>
+    });
+
+    it('should only accept maximum params of 5', () => {
+      const a = of(1, 2, 3);
+      const b = ['a', 'b', 'c'];
+      const c = ['d', 'e', 'f'];
+      const d = ['g', 'h', 'i'];
+      const e = ['j', 'k', 'l'];
+      const f = ['m', 'n', 'o'];
+      const g = ['p', 'q', 'r'];
+      const res = a.pipe(combineLatest(b, c, d, e, f, g)); // $ExpectError
+    });
+  });
+
+  describe('with project parameter', () => {
+    it('should infer correctly with project param', () => {
+      const a = of(1, 2, 3);
+      const res = a.pipe(combineLatest(v1 => 'b')); // $ExpectType Observable<string>
+    });
+
+    it('should infer correctly with 1 param', () => {
+      const a = of(1, 2, 3);
+      const b = ['a', 'b', 'c'];
+      const res = a.pipe(combineLatest(b, (a, b) => b)); // $ExpectType Observable<string>
+    });
+
+    it('should infer correctly with 2 params', () => {
+      const a = of(1, 2, 3);
+      const b = ['a', 'b', 'c'];
+      const c = ['d', 'e', 'f'];
+      const res = a.pipe(combineLatest(b, c, (a, b, c) => b + c)); // $ExpectType Observable<string>
+    });
+
+    it('should infer correctly with 3 params', () => {
+      const a = of(1, 2, 3);
+      const b = ['a', 'b', 'c'];
+      const c = ['d', 'e', 'f'];
+      const d = ['g', 'h', 'i'];
+      const res = a.pipe(combineLatest(b, c, d, (a, b, c, d) => b + c)); // $ExpectType Observable<string>
+    });
+
+    it('should infer correctly with 4 params', () => {
+      const a = of(1, 2, 3);
+      const b = ['a', 'b', 'c'];
+      const c = ['d', 'e', 'f'];
+      const d = ['g', 'h', 'i'];
+      const e = ['j', 'k', 'l'];
+      const res = a.pipe(combineLatest(b, c, d, e, (a, b, c, d, e) => b + c)); // $ExpectType Observable<string>
+    });
+
+    it('should infer correctly with 5 params', () => {
+      const a = of(1, 2, 3);
+      const b = ['a', 'b', 'c'];
+      const c = ['d', 'e', 'f'];
+      const d = ['g', 'h', 'i'];
+      const e = ['j', 'k', 'l'];
+      const f = ['m', 'n', 'o'];
+      const res = a.pipe(combineLatest(b, c, d, e, f, (a, b, c, d, e, f) => b + c)); // $ExpectType Observable<string>
+    });
+
+    it('should infer correctly with array param', () => {
+      const a = of(1, 2, 3);
+      const b = ['a', 'b', 'c'];
```

Comment:
Given that you are testing the acceptance of an array here, I think all other arrays in the other tests should be replaced with `of`:

```ts
const a = of(1, 2, 3);
const b = of('a', 'b', 'c');
const res = a.pipe(combineLatest(b)); // $ExpectType Observable<[number, string]>
```

Those tests use arrays as `ObservableInput`s - as does this test.

The more important array-related signatures are the ones that take [an array of `ObservableInput`s](https://github.com/ReactiveX/rxjs/blob/6.3.3/src/internal/operators/combineLatest.ts#L35-L38). And it should be tested with something like this:

```ts
it('should infer correctly with array param', () => {
  const a = of(1, 2, 3);
  const b = [of('x')];
  const res = a.pipe(combineLatest(b, (a, b) => b)); // $ExpectType Observable<string>
});
```

Note that `a.pipe(combineLatest([of('x'), of('y')]))` should be equivalent to `a.pipe(combineLatest(of('x'), of('y')))` and `a.pipe(combineLatest(...[of('x'), of('y')]))`.

Also, looking at the order of the signatures, I wouldn't be surprised if this test exposes a bug.